src/bin/dual-iir.rs

```diff
@@ -310,6 +344,20 @@ const APP: () = {
         c.resources.afes.0.set_gain(settings.afe[0]);
         c.resources.afes.1.set_gain(settings.afe[1]);
 
+        // Update the signal generators
+        c.resources.signal_generator.lock(|generator| {
```

Comment:
To halve the lock time (and thus limit the `process()` delay), I'd move the ref out of the `c.resources` struct and then `lock` it within each iteration of the loop.